py/desispec/sky.py

```diff
@@ -233,55 +231,30 @@ def qa_skysub(param, frame, skymodel, quick_look=False):
         qadict: dict of QA outputs
           Need to record simple Python objects for yaml (str, float, int)
     """
-    log=get_logger()
-
-    # Output dict
-    qadict = {}
-    qadict['NREJ'] = int(skymodel.nrej)
-
-    # Grab sky fibers on this frame
-    skyfibers = np.where(frame.fibermap['OBJTYPE'] == 'SKY')[0]
-    assert np.max(skyfibers) < 500  #- indices, not fiber numbers
-    nfibers=len(skyfibers)
-    qadict['NSKY_FIB'] = int(nfibers)
+    from desispec.qa import qa_quicklook as qa
+    import copy
 
-    current_ivar=frame.ivar[skyfibers].copy()
-    flux = frame.flux[skyfibers]
-
-    # Subtract
-    res = flux - skymodel.flux[skyfibers] # Residuals
-    res_ivar = util.combine_ivar(current_ivar, skymodel.ivar[skyfibers])
-
-    # Chi^2 and Probability
-    chi2_fiber = np.sum(res_ivar*(res**2),1)
-    chi2_prob = np.zeros(nfibers)
-    for ii in range(nfibers):
-        # Stats
-        dof = np.sum(res_ivar[ii,:] > 0.)
-        chi2_prob[ii] = scipy.stats.chisqprob(chi2_fiber[ii], dof)
-    # Bad models
-    qadict['NBAD_PCHI'] = int(np.sum(chi2_prob < param['PCHI_RESID']))
-    if qadict['NBAD_PCHI'] > 0:
-        log.warning("Bad Sky Subtraction in {:d} fibers".format(
-                qadict['NBAD_PCHI']))
-
-    # Median residual
-    qadict['MED_RESID'] = float(np.median(res)) # Median residual (counts)
-    log.info("Median residual for sky fibers = {:g}".format(
-        qadict['MED_RESID']))
-
-    # Residual percentiles
-    perc = dustat.perc(res, per=param['PER_RESID'])
-    qadict['RESID_PER'] = [float(iperc) for iperc in perc]
-
-    #- Add per fiber median residuals
-    qadict["MED_RESID_FIBER"]=np.median(res,axis=1)
-
-    #- Evaluate residuals in wave axis for quicklook
-    if quick_look:
-        
-        qadict["MED_RESID_WAVE"]=np.median(res,axis=0)
-        qadict["WAVELENGTH"]=frame.wave
-        qadict["SKY_FIBERID"]=skyfibers.tolist()
-    # Return
+    log=get_logger()
+    #- sky residual
+    if param is None:
+        log.info("Param is None. Using default param instead")
+        param = dict(
+                     PCHI_RESID=0.05, # P(Chi^2) limit for bad skyfiber model residuals
+                     PER_RESID=95.,   # Percentile for residual distribution
+                     )
+    #qadict_resid = qa.sky_resid(param, frame, skymodel, quick_look=quick_look)
+    #qadict=qadict_resid.copy()
+
+    #- QAs
+    #- first subtract sky to get the sky subtracted frame. This is only for QA. Pipeline does it separately. 
+    tempframe=copy.deepcopy(frame) #- make a copy so as to propagate frame as it is
```

Comment:
Again, we should avoid deepcopy()

It would be better to add a copy() method to the Frame class.